sklearn/linear_model/_quantile.py

```diff
@@ -0,0 +1,261 @@
+                f"with scipy>=1.6.0, got {sp_version}"
+            )
+
+        if self.solver_options is not None and not isinstance(
+            self.solver_options, dict
+        ):
+            raise ValueError(
+                f"Invalid value for argument solver_options, "
+                f"must be None or a dictionary, got "
+                f"{self.solver_options}"
+            )
+
+        # make default solver more stable
+        if self.solver_options is None and self.solver == "interior-point":
+            solver_options = {"lstsq": True}
+        else:
+            solver_options = self.solver_options
+
+        # Use linear programming formulation of quantile regression
+        #     min_x c x
+        #           A_eq x = b_eq
+        #                0 <= x
+        # x = (s0, s, t0, t, u, v) = slack variables
+        # intercept = s0 + t0
+        # coef = s + t
+        # c = (alpha * 1_p, alpha * 1_p, quantile * 1_n, (1-quantile) * 1_n)
+        # residual = y - X@coef - intercept = u - v
+        # A_eq = (1_n, X, -1_n, -X, diag(1_n), -diag(1_n))
+        # b_eq = y
+        # p = n_features + fit_intercept
+        # n = n_samples
+        # 1_n = vector of length n with entries equal one
+        # see https://stats.stackexchange.com/questions/384909/
+        #
+        # Filtering out zero samples weights from the beginning makes life
+        # easier for the linprog solver.
+        mask = sample_weight != 0
+        n_mask = int(np.sum(mask))  # use n_mask instead of n_samples
+        c = np.concatenate(
+            [
+                np.full(2 * n_params, fill_value=alpha),
+                sample_weight[mask] * self.quantile,
+                sample_weight[mask] * (1 - self.quantile),
+            ]
+        )
+        if self.fit_intercept:
+            # do not penalize the intercept
+            c[0] = 0
+            c[n_params] = 0
+
+            A_eq = np.concatenate(
+                [
+                    np.ones((n_mask, 1)),
+                    X[mask],
+                    -np.ones((n_mask, 1)),
+                    -X[mask],
+                    np.eye(n_mask),
+                    -np.eye(n_mask),
+                ],
+                axis=1,
+            )
+        else:
+            A_eq = np.concatenate(
+                [X[mask], -X[mask], np.eye(n_mask), -np.eye(n_mask)], axis=1
+            )
+
+        b_eq = y[mask]
+
+        result = linprog(
+            c=c,
+            A_eq=A_eq,
+            b_eq=b_eq,
+            method=self.solver,
+            options=solver_options,
+        )
+        solution = result.x
+        if not result.success:
+            failure = {
+                1: "Iteration limit reached.",
+                2: "Problem appears to be infeasible.",
+                3: "Problem appears to be unbounded.",
+                4: "Numerical difficulties encountered.",
+            }
+            raise ValueError(
```

Comment:
This should be a `RunTimeError` I think

Reply:
I changed to throwing a warning `ConvergenceWarning` instead of raising an error.